packages/core/src/fhir-to-cda/cda-templates/components/family-history.ts

```diff
@@ -0,0 +1,308 @@
+  const ageString = `${age?.value ?? ""}${age?.unit ?? ""}`;
+  return ageString.length > 0 ? ageString : undefined;
+}
+
+function getNotes(note: Annotation[] | undefined): string | undefined {
+  const combinedNotes = note?.map(note => note.text).join("; ");
+  return combinedNotes?.length ? combinedNotes : undefined;
+}
+
+function createEntryFromMemberHistory(
+  augHistory: AugmentedFamilyMemberHistory,
+  referenceId: string
+): ObservationOrganizer {
+  return {
+    organizer: {
+      _classCode: "CLUSTER",
+      _moodCode: "EVN",
+      templateId: buildInstanceIdentifier({
+        root: augHistory.typeOid,
+        extension: extensionValue2015,
+      }),
+      id: buildInstanceIdentifier({
+        root: placeholderOrgOid,
+        extension: augHistory.resource.id,
+      }),
+      statusCode: {
+        _code: mapFamilyHistoryStatusCode(augHistory.resource.status),
+      },
+      subject: {
+        relatedSubject: {
+          code: mapRelationship(augHistory.resource.relationship, referenceId),
+          subject: buildSubject(augHistory.resource),
+        },
+      },
+      component: buildComponents(augHistory.resource.condition, referenceId),
+    },
+  };
+}
+
+function buildSubject(memberHist: FamilyMemberHistory): Subject {
+  const genderCode = buildCodeCvFromCodeableConcept(memberHist.sex);
+  const mappedGenderCode = remapGenderCode(genderCode);
+
+  const birthTime = withNullFlavor(formatDateToCdaTimestamp(memberHist.bornDate), "_value");
+
+  const deceasedBoolean = memberHist.deceasedBoolean;
+  const deceasedInd = deceasedBoolean
+    ? {
+        _value: memberHist.deceasedBoolean,
+        [_xmlnsSdtcAttribute]: "urn:hl7-org:sdtc",
+      }
+    : undefined;
+
+  return {
+    name: memberHist.name,
+    administrativeGenderCode: mappedGenderCode,
+    birthTime,
+    deceasedInd, // TODO: Validator not accepting this even though this looks correct based on the spec..
+  };
+}
+
+function mapRelationship(
+  relationship: CodeableConcept | undefined,
+  referenceId: string
+): CdaCodeCv | undefined {
+  const relationshipCoding = relationship?.coding?.[0];
+  if (!relationshipCoding) return undefined;
+
+  const { code, display, system } = relationshipCoding;
+  if (!code || !system) return undefined;
+
+  const codeCe: Partial<CdaCodeCe> = {
+    _code: code,
+  };
+
+  if (system?.toLowerCase().includes("rolecode")) {
+    codeCe._codeSystem = "2.16.840.1.113883.5.111";
+    codeCe._codeSystemName = "RoleCode";
+  } else if (system?.toLowerCase().includes("snomed")) {
+    codeCe._codeSystem = snomedCodeSystem;
+    codeCe._codeSystemName = snomedSystemName;
+  } else {
+    throw new Error("Relationship code must be in SNOMED or HL7 RoleCode coding system");
+  }
+
+  return {
+    ...codeCe,
+    ...(display && { _displayName: display }),
+    originalText: buildOriginalTextReference(referenceId),
+  };
+}
+
+function remapGenderCode(genderCode: CdaCodeCe | undefined): CdaCodeCe | undefined {
```

Comment:
nit: this function name seems like it doesnt describe whats happenng. Isnt it just building the gender code component. It isnt remapping beyond what mapGenderCode is doing

Reply:
💯 ! probably a left over during development..